packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -416,25 +416,48 @@ export const createWebpackUtils = (
     } = {}): RuleSetRule => {
       return {
         test: /\.(js|mjs|jsx)$/,
-        include: (modulePath: string): boolean => {
-          // when it's not coming from node_modules we treat it as a source file.
-          if (!vendorRegex.test(modulePath)) {
-            return true
-          }
-
-          // If the module uses Gatsby as a dependency
-          // we want to treat it as src so we can extract queries
-          return modulesThatUseGatsby.some(module =>
-            modulePath.includes(module.path)
-          )
-        },
-        type: `javascript/auto`,
-        use: [
-          loaders.js({
-            ...options,
-            configFile: true,
-            compact: PRODUCTION,
-          }),
+        oneOf: [
+          /**
+           * Handle page templates separately to scope certain babel transforms appropriately
+           * (e.g. removing special module exports named `config` and `getServerData` in
+           * `babel-plugin-remove-api`).
+           */
+          {
+            resourceQuery: /page-template/,
+            type: `javascript/auto`,
+            use: [
+              loaders.js({
+                ...options,
+                compact: PRODUCTION,
```

Comment:
keep it the same as below
```suggestion
                configFile: true,
                compact: PRODUCTION,
```